app/controllers/mechanics_controller.rb

```diff
@@ -0,0 +1,9 @@
+class MechanicsController < ApplicationController
+
+  def show
+    @mechanic = Mechanic.find(params[:id])
+    @rides = Ride.joins(:mechanic_rides).where("mechanic_id = #{@mechanic.id}")
```

Comment:
You can instead just do `mechanic.rides` because you have the AR association between these 2 set up